kong/router/atc_compat.lua

```diff
@@ -428,12 +429,93 @@ function _M.new(routes, cache, cache_neg)
       routes = routes_t,
       services = services_t,
       fields = inst:get_fields(),
-      cache = cache,
-      cache_neg = cache_neg,
     }, _MT)
 end
 
 
+local function is_route_changed(a, b)
+  return a.updated_at ~= b.updated_at
+end
+
+
+local function new_from_previous(routes, is_traditional_compatible, pre_router)
```

Comment:
```suggestion
local function new_from_previous(routes, is_traditional_compatible, old_router)
```

`old_` is probably a better prefix than `pre_`.